hbase-server/src/main/java/org/apache/hadoop/hbase/master/snapshot/SnapshotHFileCleaner.java

```diff
@@ -64,8 +67,10 @@ public class SnapshotHFileCleaner extends BaseHFileCleanerDelegate {
 
   @Override
   public Iterable<FileStatus> getDeletableFiles(Iterable<FileStatus> files) {
+    List<FileStatus> filesList =
```

Comment:
Better add some comments here to say why we want a List instead of a Iterable here.

Reply:
Sure, good point!

Reply:
> The Iterable is lazy evaluated, so if we just pass this Iterable in, we will access the HFile storage inside the snapshot lock, which could take a lot of time(for example, several seconds), and block all other operations, especially other cleaners.
> So here we convert it to List first, to force it evaluated before calling getUnreferencedFiles, so we will not hold snapshot lock for a long time.

Something like this?

Thanks.